tools/CMakeLists.txt

```diff
@@ -282,3 +282,30 @@ if(IREE_BUILD_COMPILER)
     COMPONENT Tests
   )
 endif(IREE_BUILD_COMPILER)
+
+# Import the RuntimeImpl target when cross compiling.
+# TODO(scotttodd): Refactor to import_shared_library/import_executable funcs?
+set(_RUNTIME_IMPL_NAME "iree_compiler_API2_RuntimeImpl")
+set(_RUNTIME_IMPL_ALIAS "iree::compiler::API2::RuntimeImpl")
+if(TARGET "${_RUNTIME_IMPL_NAME}")
+  # Already defined, nothing to do.
+elseif(DEFINED IREE_HOST_BINARY_ROOT)
+  # Import.
+  set(_FULL_LIBRARY_NAME "${CMAKE_SHARED_LIBRARY_PREFIX}IREECompiler${CMAKE_SHARED_LIBRARY_SUFFIX}")
+  set(_LIBRARY_BIN_PATH "${IREE_HOST_BINARY_ROOT}/bin/${_FULL_LIBRARY_NAME}")
+  set(_LIBRARY_LIB_PATH "${IREE_HOST_BINARY_ROOT}/lib/${_FULL_LIBRARY_NAME}")
+  if(EXISTS ${_LIBRARY_BIN_PATH})
+    add_library("${_RUNTIME_IMPL_NAME}" SHARED IMPORTED GLOBAL)
+    set_property(TARGET "${_RUNTIME_IMPL_NAME}" PROPERTY IMPORTED_LOCATION "${_LIBRARY_BIN_PATH}")
+    add_library("${_RUNTIME_IMPL_ALIAS}" ALIAS "${_RUNTIME_IMPL_NAME}")
```

Comment:
We could consider replacing the `iree_get_executable_path` implementation with this as well.

If `IREE_HOST_BINARY_ROOT` (I can rename this to `IREE_HOST_INSTALL_ROOT`) is set, we'd import `iree-compile` and the other binary targets. We could then have a single place that imports all targets (shared libraries and executables) from that install path.

Reply:
What if we move the logic for building from source vs using prebuilt from all the targets using installable artifacts to the producers of them instead. So anything that uses `iree-compile` can depend on the target and trust that everything Just Works:tm:. I think all the logic can be in `iree_cc_binary` (we currently install all binaries, though I think we should probably stop doing that, since some are just random tests). To support the shared libraries, the binary would need to inspect its dependencies to see if they're shared, and if so add a dep on the appropriate file. Ah or rather than that (which sounds like a lot of introspection) it can just keep deps if they're a shared library and `iree_cc_library` can do similar logic to look up its prebuilt version if it's shared. WDYT? I think something like that would work pretty well.

Reply:
Yeah, I have something like this in mind, but my brain is fried from all this CMake work - I might pick it up later.

```cmake
if(IREE_BUILD_COMPILER)
  iree_cc_binary(iree-compile ...)
  iree_cc_binary(iree-opt ...)
elseif(IREE_HOST_BINARY_ROOT)
  # add IMPORTED libraries/binaries
  iree_import_binary(iree-compile
    SHARED_LIBRARY_DEPS
      IREECompiler
  )
  iree_import_binary(iree-opt
    SHARED_LIBRARY_DEPS
      IREECompiler
  )
else()
  # skip defining iree-compile and other targets
endif()

# elsewhere just use the iree-compile target, ask it for TARGET_FILE using generator expressions, etc.
```

Reply:
(`SHARED_LIBRARY_DEPS IREECompiler` there could then also be imported with `iree_import_shared_library` to the `iree::compiler::API2::Impl` name, or something similar enough)